jflex/src/main/java/jflex/LexGenerator.java

```diff
@@ -0,0 +1,131 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * JFlex 1.7.1-SNAPSHOT                                                    *
+ * Copyright (C) 1998-2018  Gerwin Klein <lsf@jflex.de>                    *
+ * All rights reserved.                                                    *
+ *                                                                         *
+ * License: BSD                                                            *
+ *                                                                         *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+package jflex;
+
+import static jflex.Options.encoding;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * This is the main class of JFlex controlling the scanner generation process. It is responsible for
+ * parsing the commandline, getting input files, starting up the GUI if necessary, etc.
```

Comment:
The comment will need adjustment for the refactor, indicating which part is in which class in the new setting.